### PR TITLE
Allow all classes for @apply

### PR DESCRIPTION
**Pull request by garymathews**
Relates to https://github.com/tailwindlabs/tailwindcss/issues/6321

- Basically, _apply_ only works for Tailwind generated classes under JIT; which is default when moving to Tailwind 3.
  - This issue is also present in Tailwind 2 when using JIT.

### CODE REVIEW

File: src/lib/expandApplyAtRules.js [hunk 1]
```diff
@@ -122,6 +122,27 @@ function processApply(root, context) {
...
+      if (node.selector && node.selector[0] === '.') {
+        const name = node.selector.substring(1)
+
+        // Ignore unused candidates.
+        if (applyCandidates.has(name) && !applyClassCache.has(name)) {
+          applyClassCache.set(name, [
+            [
+              {
+                sort: BigInt(context.layerOrder.user),
+                layer: 'user',
```
Comment (garymathews) [inline src/lib/expandApplyAtRules.js:136]: I'm not too sure about these

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

If I run your tests against the current `master` branch, then it already passes. I'll look into the actual issue and mark you as a contributor if I have a fix.

**Comment by garymathews** [comment]
@RobinMalfait, thanks for fixing this 👍  https://github.com/tailwindlabs/tailwindcss/pull/6589

However, I noticed it's required to reference `@tailwind;` to the particular css file with the `@apply` for this to work (don't even need to define a directive); where before this was not necessary. (I should only need to define directives in my main css file)

```JS
it('should pickup all classes', () => {
  let config = {
    content: [{ raw: html`<div class="foo"></div>` }],
    plugins: [],
  }

  let input = css`
    @tailwind; /* remove me I break */

    .bop {
      color: red;
    }
    .bar {
      background-color: blue;
    }
    .foo {
      @apply absolute bar bop;
    }
  `

  return run(input, config).then((result) => {
    return expect(result.css).toMatchFormattedCss(css`
      @tailwind;
      .bop {
        color: red;
      }
      .bar {
        background-color: blue;
      }
      .foo {
        position: absolute;
        color: red;
        background-color: blue;
      }
    `)
  })
})
```
```
The `bar` class does not exist. If `bar` is a custom class, make sure it is defined within a `@layer` directive.
```

**Comment by garymathews** [comment]
I found its due to an optimization where css files without a directive are ignored:
[setupTrackingContext.js#L127](https://github.com/tailwindlabs/tailwindcss/blob/master/src/lib/setupTrackingContext.js#L127)
[setupWatchingContext.js#L244](https://github.com/tailwindlabs/tailwindcss/blob/master/src/lib/setupWatchingContext.js#L244)

**Comment by garymathews** [comment]
@RobinMalfait I've updated my PR with a solution that maintains the optimization for css without @tailwind or @apply

**Pull request merged**